frontend/views/pages/PayGroup.vue

```diff
@@ -20,65 +27,205 @@ page(
+
+  section.card
+    nav.tabs(
+      v-if='!needsIncome'
+      aria-label='navigation'
+      @click='open = false'
+    )
+      a.tabs-link(
+        v-for='(link, index) in tabItems'
         :key='index'
+        :class='{ "tabs-link-active": activeTab === link.url}'
+        :data-test='`link-${link}`'
+        @click='activeTab = link.url'
       )
-        .c-info
-          avatar-user.c-avatar(:username='payment.to' size='sm')
-          div(role='alert')
-            i18n(
-              tag='p'
-              :args='{ \
-                total: `<b>${payment.amountFormatted}</b>`, \
-                user: `<b>${payment.to}</b>` \
-              }'
-            ) {total} to {user}
-            span(
-              v-if='payment.hash'
-              :class='payment.paymentClass'
-            ) {{ payment.paymentStatusText }}
-
-        .buttons.is-start.c-ctas
-          router-link.button.is-small.is-outlined(
-            v-if='payment.data.status === statusType.PAYMENT_NOT_RECEIVED'
-            to='messages/liliabt'
-          ) {{ L('Send Message...') }}
-          i18n.button.is-small.c-ctas-send(
-            tag='button'
-            key='send'
-            v-if='showMarkAsPaid(payment)'
-            @click='markAsPayed(payment.to, payment)'
-          ) Mark as sent
-          i18n.button.is-small.is-outlined(
-            tag='button'
-            key='cancel'
-            v-if='payment.data.status === statusType.PAYMENT_PENDING'
-            @click='cancelPayment(payment.to, payment.hash)'
-          ) Cancel
-
-  .c-footer
-    i18n.button.is-small.is-outlined(
-      tag='button'
-      @click='seeHistory'
-    ) See past contributions
+        | {{ link.title }}
+        span.tabs-notification(v-if='link.notification') {{ link.notification }}
+
+    form.c-search-form(
+      v-if='needsIncome || activeTab !== "todo"'
+      @submit.prevent='submit'
+    )
+      label.field
+        i18n.sr-only Search payments
+        .input-combo
+          .is-icon(:aria-label='L("Search")')
+            i.icon-search
+          input.input(
+            type='text'
+            name='search'
+            data-test='search'
+            placeholder='Search payments...'
+            v-model='form.search'
+          )
+          button.is-icon-small(
+            v-if='form.search'
+            aria-label='Clear search'
+            @click='form.search = null'
+          )
+            i.icon-times
+
+    section.tab-section
+      .c-container(v-if='hasPayments')
+        table.table.table-in-card.c-payments
+          thead
+            tr(v-if='activeTab === "todo"')
+              i18n(tag='th') Send to
+              i18n.c-payments-amount(tag='th') Amount
+              i18n.c-payments-date(tag='th') Due in
+            tr(v-else)
+              i18n(tag='th') Sent to
+              i18n.c-payments-amount(tag='th') Amount
+              i18n.c-payments-date(tag='th') Date
```

Comment:
Heres a suggestion to get rid of this ternary that duplicates the markup twice:

```js
tr
  th {{ tableTitles.one }}
  th.c-payments-amount {{ tableTitles.two }}
  th.c-payments-date {{ tableTitles.three }}

// computed property
tableTitles () {
  return this.activeTab === "todo" ? {
    one: 'Send to', // with L() obviously.
    two: 'Amount',
    three: 'Due in'
  } : {
    one: 'Sent to'
    two: 'Amount'
    three: 'Date'
  }
}
```

Personally I like to keep the markup as clean as possible and deal with the logic (in this case text logic) on the JS side. What do you think?

Reply:
I personally don't think that "th {{ tableTitles.one }}" is much easier to read, but sure